### PR TITLE
BUGFIX: Do not throw an exception if the shortcut node could not be resolved

### PR DESCRIPTION
**Pull request by daniellienert**
Now specific log messages are written when the target could not be resolved and leads to 404 in the frontend.

Fixes: #3256

### CODE REVIEW

**Review by markusguenther — APPROVED** [review]
Thanks Daniel, looks good by reading but need to test it 👨‍💻

**Review by bwaidelich — COMMENTED** [review]
Sorry for messing this up and thanks a lot for the fix.

I need to take a closer look at this but for the moment I would suggest to keep the fix as minimal invasive as possible, i.e. don't add a new logging feature with it (at least not in the same PR).

Will test asap

**Comment by markusguenther** [comment]
Thanks @bwaidelich for the valuable feedback :)
Guess will test it until mid of the week.

File: Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php [hunk 1]
```diff
@@ -303,18 +303,19 @@ protected function truncateUriPathSuffix(string $uriPath)
...
     protected function resolveShortcutNode(NodeInterface $node)
     {
         $resolvedNode = $this->nodeShortcutResolver->resolveShortcutTarget($node);
         if (is_string($resolvedNode)) {
             return new Uri($resolvedNode);
         }
+
         if (!$resolvedNode instanceof NodeInterface) {
-            throw new NeosException(sprintf('Could not resolve shortcut target for node "%s"', $node->getPath()), 1414771137);
+            return $node;
```
[the remark this answers is not in the document]
Comment (daniellienert) [reply Neos.Neos/Classes/Routing/FrontendNodeRoutePartHandler.php:316]: Agreed. Catching the exception is what happened in previous versions.

About adding logging: Previously the exception was logged. Adding a manual logging here would be the only way to find redirects that are not valid anymore.

**Comment by bwaidelich** [comment]
@daniellienert Thanks for your feedback, I'll look into it later today or tomorrow

**Comment by bwaidelich** [comment]
This has been fixed with #3269 in the meantime. Please re-open/comment if I mixed things up

**Pull request closed without merging**